Support multiple compose files

mathieu-lemay opened the pull request:
This is something I needed for a project at work, so I hacked this solution quickly. I'm fully open to improvement suggestions.

RmStorm commented:
Hi @mathieu-lemay , sorry that I didn't respond earlier! This looks good and as far as I can tell it's fully backwards compatible? If you can write a little documentation with an example of how to use multiple files and a testcase that uses multiple files I'll gladly merge this.

mathieu-lemay commented:
@RmStorm No worries! I'm currently on vacation with pretty limited internet access, but I'll update the PR with tests when I'm back.

As for the backward compatibility, I haven't explicitly tried it but it should be, so it won't break any existing workflow.

mathieu-lemay commented:
@RmStorm There was a little issue with python3.5 not supporting Path objects for the os.path.* functions, so I fixed that and all the existing tests are now passing. Also added one test to validate the new functionality and an example in the readme.

todofixthis commented on README.rst:
```diff
@@ -167,6 +167,12 @@ working directory.  You can specify a different file via the
 
     pytest --docker-compose=/path/to/docker-compose.yml
 
+To specify more than one compose file, separate them with a ``,``:
```
❔ Could this be expanded a bit, to explain the result of specifying multiple compose files, and maybe when it could be useful to do this?

RmStorm replied:
I agree it's a relatively unknown feature (I think?) So a quick link to the docker-compose documentation where it explains how multiple files work is definitely nice!
https://docs.docker.com/compose/extends/

RmStorm commented:
> @RmStorm There was a little issue with python3.5 not supporting Path objects for the os.path.* functions, so I fixed that and all the existing tests are now passing. Also added one test to validate the new functionality and an example in the readme.

yeah, that's a little annoying. Not much to do about it I think. Changing everything to strings instead of Paths would not be much of an improvement...

RmStorm commented:
I think this looks good. I will merge it to dev, update the other dependencies and add the link to the compose documentation and then merge to master.
Thanks @mathieu-lemay !

mathieu-lemay commented:
> yeah, that's a little annoying. Not much to do about it I think. Changing everything to strings instead of Paths would not be much of an improvement...

Agreed, especially since I need the Path objects for the `relative_to` method. If/when support for python3.5 is dropped, those two lines can be removed though.

Also, thanks for adding the link for the documentation.
